readthedocs/api/v3/serializers.py

```diff
@@ -222,13 +222,23 @@ def get_project(self, obj):
         return self._absolute_url(path)
 
 
-class VersionURLsSerializer(serializers.Serializer):
+class VersionURLsSerializer(BaseLinksSerializer, serializers.Serializer):
     documentation = serializers.SerializerMethodField()
+    edit = serializers.SerializerMethodField()
```

Comment:
Shouldn't this be called `detail` to keep the same naming than the view?

Reply:
Good point. It's not a detail view though, it's a form view. I think the view is probably just poorly named.

Reply:
Should we use `update` instead perhaps? `edit` is still probably a bit clearer than `update`, but i don't have strong opinions

Reply:
I think we have list (eg. `/objects/`, or https://readthedocs.org/projects/docs/versions/) and detail views (eg. `/objects/<pk>/`, https://readthedocs.org/dashboard/docs/version/latest/) --which is how Django REST Framework calls them (the action is communicated by the HTTP verb). I don't think we should have `edit` or `update` views.

For example, https://readthedocs.org/dashboard/docs/edit/, is not correct to me and it should be something like https://readthedocs.org/dashboard/docs/admin/, where all the other views to edit/update something else related to the project are nested under `/admin/`. Then, https://readthedocs.org/dashboard/docs/domains/ should be https://readthedocs.org/dashboard/docs/admin/domains/

So, I wouldn't try to replicate the `/edit` postfix pattern here as well because I don't think it's correct. IMHO, in our case for edit version, it should be called `detail`. So, under the version's detail page, you can see all the details of the versions and perform some actions as well (eg. edit) which will be translated in the proper HTTP verb.

Reply:
I'd probably disagree on edit vs detail distinction here. We use this as an edit view in both our current theme and it is strictly an edit view in the new theme. That is, there is no view that users go to see the details of a version, admin users go to this view to edit the version. We really only have maybe 2 detail views -- build detail and project detail. Ultimately, it will be the edit view URL that I need, to point users from the version "create" view to the update view of the selected version.

I'm not sure we want to move version edit back under the admin dashboard again. Either way, we use the same pattern under admin dashboard pages -- `/edit/` postfix on URLS -- for edit views on model objects. This was copying that pattern.

So for this use case `detail = serializers.SerializerMethodField()` doesn't seem correct

Reply:
I prefer to follow the conventions that are established already in Django and DRF. As I mentioned before, there are `list` and `detail` URLs (which maps to the template suffix file: `_detail.html` and `_list.html`); then the actions (e.g. create, update) are HTTP verbs --which can be done from from those templates. Usually, create/POST is done from the `_list.html` and update/PATCH is done from `_detail.html` one.

Considering this is why prefer to keep calling `list` and `detail` URLs (without adding the `/edit/` suffix to URLs as done in #7595).

Depending on each object in particular, the `detail` URL may allow you to edit the object or not. The same happens with the `list` URL, where you may not be allowed to create new objects from that listing.

Reply:
Roger, I think the confusion here might be between DRF patterns and standard Django patterns. It's a bit different in django vanilla views and django based views.

A `list` view is used via `GET` and create/POST is not performed from this view. `create` is the form that creates/POST the object (not applicable to this model/view), `detail` is a view that only *displays* information about the model instance, and `edit` is a view that *displays editable* information about the model instance.

The current/future version view is not a detail view -- it's a form view and that is really only useful for admin users. Our other URLs on the project admin dashboard all use `../edit/` in the URL. I think this is correct and this URL change follows that pattern.

Reply:
Trying to make sense of outstanding PRs that I'm blocked on, I realized that the underlying for `/edit/` is not clear. I added more context here: https://github.com/readthedocs/readthedocs.org/pull/7595#issuecomment-812731544

It does seem the conflict here is DRF naming conventions vs Django URL patterns. I have matched what we use in our dashboard in other views, so feel safe with this naming convention. The underlying change is because a URL without an additional verb or separator conflicts with the URL for ProjectVersionCreate.